feat: Add type annotations and refactor test utilities

Lumabots opened the pull request:
Added explicit type annotations to test functions and variables for improved type safety and clarity. Refactored test predicates in test_find_util.py to named functions for better readability. Updated imports and variable declarations to use modern typing and collections. Minor cleanup in test_markdown_utils.py to remove unused imports.

## Summary

<!-- What is this pull request for? Does it fix any issues? -->

## Information

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [ ] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

VincentRPS commented:
This PR is now usable. All that is needed is fixing a few weird bugs and implementing the rest of currently supported events

VincentRPS commented:
I am going to do a marginal redesign to fit https://github.com/Pycord-Development/pycord-v3/discussions/128 and give more royalty to the caching system

Lumabots committed:
Add type annotations and refactor test utilities

Added explicit type annotations to test functions and variables for improved type safety and clarity. Refactored test predicates in test_find_util.py to named functions for better readability. Updated imports and variable declarations to use modern typing and collections. Minor cleanup in test_markdown_utils.py to remove unused imports.

Paillat-dev commented on tests/test_find_util.py:
```diff
@@ -22,69 +22,106 @@
+    return x > 3
+
+
+def equals_1(x: int) -> bool:
```
All of the equals_... should return `TypeIs[Literal[what they return]]`. Import `TypeIs` from `typing_extensions`.

Paillat-dev commented on tests/test_format_dt.py:
```diff
@@ -25,14 +25,23 @@
 random.seed(42)
 
-ALL_STYLES = ["t", "T", "d", "D", "f", "F", "R", None]
+ALL_STYLES: list[TimestampStyle | None] = [
```
No need to type a constant since it never changes.

Paillat-dev replied:
Same for all caps constant, dont type them it can actually make the typing less accurate

Lumabots committed:
refactor: simplify type annotations and improve test utility definitions

Lumabots committed:
fix: update test for generate_snowflake to directly call function

Paillat-dev commented on tests/test_find_util.py:
```diff
@@ -22,69 +22,109 @@
+    return x == 30
+
+
+def is_none_pred(x: object) -> bool:
```
Here als
```suggestion
def is_none_pred(x: object) -> TypeIs[Literal[None]]:
```

Paillat-dev committed:
Update tests/test_find_util.py

Signed-off-by: Paillat <jeremiecotti@ik.me>
